### PR TITLE
Added pagination to the Movie list page

### PR DESCRIPTION
**Pull request by shobanamg**
# Description
Added Pagination
Adjusted search, sortby, and category filter components regard to pagination.
Added Loader component.
Fixed the problem of repeating movies when we paginate.
Fixed sortby design.

# Jira task link
https://hackyourfuture-dk.atlassian.net/browse/LIST-42

# Screenshots (if applicable)
movieList

### CODE REVIEW

**Comment by mohamedaldahoul** [comment]
@shobanamg the screenshot you added shows the filtering based on the category, not the pagination

**Comment by mohamedaldahoul** [comment]
I see, could you fix the conflict, we need help someone to review the Backend changes, the frontend part looks good.

**Review by itsgauravsaxena — APPROVED** [review]
BE changes looks fine too

File: packages/client/src/context/movieListContext.js [hunk 1]
```diff
@@ -46,38 +60,72 @@ export const MovieListProvider = ({ isFavoritePage, children }) => {
...
+    setSortBy(value);
+  }, []);
+
+  const onChangeDirection = useCallback(
+    (value) => {
+      setCurrentPage(1);
+      setMovies([]);
+      sortBy === value
+        ? setIsClickedSame((prevState) => !prevState)
+        : setIsClickedSame(false);
```
[the remark this answers is not in the document]
Comment (shobanamg) [reply packages/client/src/context/movieListContext.js]: It makes sense, we will try to rename it properly, maybe in the next PR

**Pull request merged**